### DISCUSSION TITLE
Auto left click on hovering elements

### DISCUSSION THREAD

**Discussion by N3C2L** [madprops/grasshopper#167] [Ideas]
I love to save mouse clicks wherever I can to save time and nerves. 😁 
  That's why I thought about interface elements that will get auto clicked on hovering.
  Ofc be disabled by default since it's a special feature, somebody will love it (like me 😁 ), someone won't like it.
  I thought about 2 important areas where auto-clicks would make most sense for me: 
  1. Auto left click on hovering loaded tabs (auto left click on unloaded tabs wouldn't make sense for me because I doesn't want to trigger them by accident)
  2. Auto left click on hovering the hover button. I think should be there since it's a hover button, and one user wants to have more hover options here.

Relevant things that occur here for users would be a setting to set the delay when the auto click on that elements should kick in.
And maybe a toggle auto click cmd where users could deactivate this auto click feature global if needed.
I'd be glad if you want to implement that 😁

**Comment by madprops** [comment]
It's a strange feature. Not sure I want to implement this, but i'll consider it.

I'm trying some initial implementation, it's not that complicated.

**Reply by N3C2L** [reply]
That's great, thank you!

**Reply by madprops** [reply]
What delay do you think would be proper?

500ms, 1 second?

500ms seems fine, I'm going to add the delay settings in `More` anyway.

**Reply by N3C2L** [reply]
Thanks, yes, 500ms is a good value and thanks for this delay settings under "More".

**Reply by madprops** [reply]
I'm going to add all these settings to a new category called `Autoclick`.

**Reply by N3C2L** [reply]
This is even better, thanks 😁

**Comment by madprops** [comment]
It's up, now I'm gonna sleep. Need to upload a fix. Done.

**Comment by N3C2L** [comment]
This new category is massive! Big up!

Many settings and also separate delay adjustments for every setting -  huge! 
I have turned on many of these autoclicks, thanks 🎉 

Some enhancements that could be useful:

1. A cmd to toggle autoclick would be good. This cmd could enable/disable all activated autoclicks settings at once (toggling on/off only the ones that a user has activated prior in the "Autoclick" category.)
But if that's not possible than a cmd to toggle on/off autoclick on tabs only would be sufficient for me, because it's the only reason I'd need that toggle cmd.

2. I'd appreciate another autoclick setting for the autoclick category: The "category" button ("General" in this pic) in the settings:

<img width="322" alt="floorp gh" src="https://github.com/user-attachments/assets/e51ae6a7-63ce-422d-b480-a2053fb8846d">

(Because that button is heavy used when jumping inside the settings from category to category)

3. I love the new context autohide setting under "More". However, I noticed that I need to be in the context menu first in order to make the autohide work. This is problematic when for example autoclicking the hover button, and then I realize that the cursor is not centered on the first item (or the first icon on compact list). You could repro that by autoclicking the hover button (compact menu enabled) and then move the mouse straight above. Result is that the context menu won't close with the context autohide setting enabled.
Another little problem also when context autohide is enabled (occuring on menus filter search), please repro: Go to Settings > Favorites > and click on any "add" button of one menu > after clicking "add" try to search something that doesn't exist in this list > problem: the menu is autoclosing itself because it's outside the context menu. Solution could be: After clicking the "add" button, the mouse cursor should be positioned on top inside this popup. Because then, the list won't get closed by itself when it doesn't find anything through the filtering process.

**Comment by madprops** [comment]
All these have been addressed in the new version.

The settings autoclick setting enables autoclick for all top settings buttons.

The context autohide should work better now with some "complicated" heuristics I added.

Global autoclick toggle and command.

Making delay default 380ms.

Adding palette autoclick.

Currently mozilla's addon server is not allowing uploads

**Comment by N3C2L** [comment]
Thanks for the updates, and the upcoming one too when the mozilla server is allowing uploads again 👍 

The autoclick toggle is great! I have it on double shift assigned.

The new autoclick setting "Settings Autoclick" is good, and thanks that you included the "Close" and the other button on top there aswell in this autoclick setting!

The context autohide is better now, thanks!

**Comment by N3C2L** [comment]
This idea has been implemented, thanks. I think I can close this.
